### PR TITLE
Feat/2023 08/add context timeouts

### PR DESCRIPTION
**Pull request by mattcataws**
*Issue #, if available:*

*Description of changes:*
Contexts were not previously plumbed through the code which didn't allow for cancelling or timeouts. This change now adds support for timeouts in the `grow` command (defaulting to 5 minutes) and passes the context through to the necessary functions. This will prevent the grow command from not exiting automatically if macOS's `diskutil` CLI happens to sleep or stop responding.

This change will also allow callers to override the timeout if needed to any duration (see [time.ParseDuration()](https://pkg.go.dev/time#ParseDuration)).

Example run with a 1 second timeout:
```
% sudo ./ec2-macos-utils_amd64 grow --id root --timeout 1s
INFO[17 Aug 23 17:32 GMT] Configuring diskutil for product              product="macOS Ventura 13.4.0"
INFO[17 Aug 23 17:32 GMT] Attempting to grow container...               device_id=disk3s4s1
INFO[17 Aug 23 17:32 GMT] Checking if device can be APFS resized...     device_id=disk3s4s1
INFO[17 Aug 23 17:32 GMT] Device can be resized                        
INFO[17 Aug 23 17:32 GMT] Repairing the parent disk...                 
INFO[17 Aug 23 17:32 GMT] Repairing parent disk...                      parent_id=disk0
Error: timeout exceeded
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,14 +1,27 @@
 module github.com/aws/ec2-macos-utils
 
-go 1.16
+go 1.21
 
 require (
 	github.com/Masterminds/semver v1.5.0
 	github.com/dustin/go-humanize v1.0.0
 	github.com/golang/mock v1.3.1
```
Comment (jahkeup) [inline go.mod:8]: Cut #8 to later replace with Uber's `gomock`

File: internal/cmd/grow_container.go [hunk 2]
```diff
@@ -38,6 +41,7 @@ with its identifier (e.g. disk1 or /dev/disk1). The string
 	growArgs := growContainer{}
 	cmd.PersistentFlags().StringVar(&growArgs.id, "id", "", `container identifier to be resized or "root"`)
 	cmd.PersistentFlags().BoolVar(&growArgs.dryrun, "dry-run", false, "run command without mutating changes")
+	cmd.PersistentFlags().DurationVar(&growArgs.timeout, "timeout", 5*time.Minute, "Set the timeout for the command (e.g. 30s, 1m, 1.5h)")
```
Comment (jahkeup) [inline internal/cmd/grow_container.go:44]: Suggestion: this magic number is pretty important to us - I suggest moving to a constant with a comment why this number is our default.

File: go.mod [hunk 3]
```diff
@@ -1,14 +1,27 @@
 module github.com/aws/ec2-macos-utils
 
-go 1.16
+go 1.21
```
Comment (jahkeup) [inline go.mod:3]: Had a quick chat with Matt - we don't expect to need 1.21 and will see if we can lower the requirement until 1.21 is more widely available/adopted.

Will post back here with an update, a bit later today.

**Review by jahkeup — COMMENTED** [review]
Small things, otherwise looking good!

File: go.mod (hunk 3 above)
Comment (mattcataws) [reply go.mod]: Done

**Review by jahkeup — APPROVED** [review]

**Review by joikawa — APPROVED** [review]

**Pull request merged**